src/pypi_changes/_print.py

```diff
@@ -32,23 +33,55 @@ def print_tree(distributions: Iterable[Package], options: Options) -> None:
         text.stylize(f"link https://pypi.org/project/{pkg.name}/#history")
         text.append(" ", "white")
         text.append(pkg.version, "blue")
-        last_release = pkg.last_release
-        if last_release is not None:  # pragma: no branch
-            if last_release["version"] != pkg.version and pkg.info is not None:
-                for a_version, releases in pkg.info["releases"].items():  # pragma: no branch
-                    first_release_at = releases[0]["upload_time_iso_8601"]
-                    if a_version == pkg.dist.version and first_release_at is not None:
-                        text.append(" ")
-                        text.append(naturaltime(now - first_release_at), "green")
-                        break
-                text.append(f" remote {last_release['version']}", "red")
-            if last_release["upload_time_iso_8601"] is not None:  # pragma: no branch
+
+        current_release = pkg.current_release
+        current_release_at = current_release.get("upload_time_iso_8601")
+        last_release = pkg.last_release or {}
+        last_release_at = pkg.last_release_at
+
+        if current_release_at is not None:
+            text.append(" ")
+            text.append(naturaltime(now - current_release_at), "green")
+        if pkg.version != last_release.get("version"):
+            text.append(f" remote {last_release.get('version')}", "red")
```

Comment:
How come we needed to change the rich layer to add the JSON presentation?

Reply:
The output hasn't changed to the best of my knowledge. This is a refactor to try and get all of the common "data analysis and aggregation" logic outside of the presentational layer so that it can be used by the various output formats (rich, JSON, and potential future additions).